tools/osbuild-mpp

```diff
@@ -1604,17 +1604,20 @@ class ManifestFileV2(ManifestFile):
             if input_count > 1:
                 raise ValueError(f"Only one of 'path', 'url' or 'text' may be specified for '{uid}'")
 
+            checksum = None
             if path:
                 f, _ = self.find_and_open_file(path, [], mode="rb", encoding=None)
                 with f:
                     data = f.read()
-                checksum = hashlib.sha256(data).hexdigest()
             elif url:
                 response = urllib.request.urlopen(url)
-                h = hashlib.file_digest(response.fp, 'sha256')
-                checksum = h.hexdigest()
+                if "file_digest" in dir(hashlib):
```

Comment:
Let's do `hasattr` here or even a `sys.version`.

Reply:
updated. I prefer to not use the version, in case something is backported or something